build.sh

```diff
@@ -356,117 +491,187 @@ EOF
+        "1" "Linux Kernel" "$BUILD_KERNEL" \
+        "2" "Clean Kernel sources" "$CLEAN_KERNEL_SRC" \
+        "3" "Allow Kernel config changes" "$ALLOW_KERNEL_CONFIG_CHANGES" \
+        "4" "Debian Rootfs" "$BUILD_ROOTFS" \
+        "5" "Allow Rootfs changes" "$ALLOW_ROOTFS_CHANGES" \
+        "6" "Ask for extra apt pkgs" "$ASK_EXTRA_PKGS" \
+        "7" "Enable ZRAM on rootfs" "$ZRAM_ENABLED" 
+        
+    # Accept user choices
+    BUILD_KERNEL='off'
+    CLEAN_KERNEL_SRC='off'
+    ALLOW_KERNEL_CONFIG_CHANGES='off'
+    BUILD_ROOTFS='off'
+    ALLOW_ROOTFS_CHANGES='off'
+    ASK_EXTRA_PKGS='off'
+    ZRAM_ENABLED='off'
+
+    [[ $selection == *1* ]] && BUILD_KERNEL='on'
+    [[ $selection == *2* ]] && CLEAN_KERNEL_SRC='on'
+    [[ $selection == *3* ]] && ALLOW_KERNEL_CONFIG_CHANGES='on'
+    [[ $selection == *4* ]] && BUILD_ROOTFS='on'
+    [[ $selection == *5* ]] && ALLOW_ROOTFS_CHANGES='on'
+    [[ $selection == *6* ]] && ASK_EXTRA_PKGS='on'
+    [[ $selection == *7* ]] && ZRAM_ENABLED='on'
+else # at least kernel or rootfs has been selected via command line, check other options and set defaults
+    [[ -z $CLEAN_KERNEL_SRC  ]] && CLEAN_KERNEL_SRC='on'
+    [[ -z $ALLOW_KERNEL_CONFIG_CHANGES  ]] && ALLOW_KERNEL_CONFIG_CHANGES='off'
+        
+    [[ -z $ALLOW_ROOTFS_CHANGES  ]] && ALLOW_ROOTFS_CHANGES='off'    
+    [[ -z $ASK_EXTRA_PKGS  ]] && ASK_EXTRA_PKGS='off'
+    [[ -z $ZRAM_ENABLED  ]] && ZRAM_ENABLED='on' 
 fi
 
-if [ ${BUILD_ROOT} = 'yes' ]
-then
-    echo "Rootfs ${release} , Hostname ${def_hostname}"
-    echo "Build initramfs: $BUILD_INITRAMFS"
-    echo "Allow rootfs changes: $ALLOW_ROOTFS_CHANGES"
+if [[ $BUILD_KERNEL == "on" ]] && [ -z "$kernel_branch" ]; then
+    display_select "Kernel Building" "Please select the Linux Kernel branch to build." \
+        "4.18" "Linux kernel 4.18" \
+        "5.6" "Linux kernel 5.6" \
+        "5.8" "Linux kernel 5.8" \
+        "5.10" "Linux kernel 5.10" \
+        "5.11" "Linux kernel 5.11" \
+        "5.12" "Linux kernel 5.12"
+        
+    ############################################################
+    # Required gcc:
+    #  armada370-gcc464_glibc215_hard_armada-GPL.txz (included in git)    FOR KERNEL VERSION <= 5.6
+    #  gcc-arm-none-eabi (downloadable via apt)         FOR KERNEL VERSION >= 5.6
+    # check toolchain subfolder for these or install via apt
+    # Adjust makehelp to match path to your gcc:
+    ############################################################
+    if [[ $selection == "4.18" ]]; then
+        makehelp='make CROSS_COMPILE=/opt/arm-marvell-linux-gnueabi/bin/arm-marvell-linux-gnueabi- ARCH=arm'   #FOR KERNEL VERSION <= 5.6 (via txz)
+        display_result "Warning" "You have selected Linux Kernel 4.18, this will be build with older armada370 gcc464 toolchain, make sure you have extracted the txz file available in toolchain folder to /opt/"
+    fi
+
+    kernel_branch="linux-$selection.y"
```

Comment:
Is this intended, that the version ends with an _y_? I saw that it works, but i think this would also create another folder in the cache directory even if the same version exists, because the folder is named differently. (linux-5.10.y and linux-5.10.39 could contain the exact same files, but the script would create the duplicate and not reuse the other)

Reply:
Well the naming with .y refers to a kernel branch: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/refs/
It tracks the kernel version without any regard to the subversion.

If you use 5.10.39 you are selecting an exact kernel version using a tag, and syntax would be 'v5.10.39' 

Both are possible and yes, both result in different cache folders. 

It might be better, to always name cache folder etc. by using the exact kernel version downloaded, need to think about this.